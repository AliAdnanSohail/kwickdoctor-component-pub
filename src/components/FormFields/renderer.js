import React from 'react';
import classnames from 'classnames';

import styles, { error as errorStyles, label as labelStyles } from './styles';

export default render => ({
  input, meta, label, ...rest
}) => (
  <div className={classnames('form-field', rest.containerClassName)}>
    <label htmlFor={rest.id}>{label}</label>
    {render(input, label, {
      ...rest,
      className: classnames({
        invalid: meta && meta.error && meta.touched,
        active: meta && meta.active,
      }),
    })}
    {meta && meta.error && meta.touched ? <div className="error">{meta.error}</div> : undefined}

    <style jsx>{styles}</style>
    <style jsx>{labelStyles}</style>
    <style jsx>{errorStyles}</style>
  </div>
);
