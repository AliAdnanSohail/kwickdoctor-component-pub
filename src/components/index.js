export { default as AvatarPicker } from './AvatarPicker';
export { default as Button } from './Button';
export { default as ButtonGroup } from './ButtonGroup';
export { default as Calendar } from './Calendar';
export { default as Card } from './Card';
export { default as CardPicker } from './CardPicker';
export { default as DatePicker } from './DatePicker';
export { default as FileInput } from './FileInput';
export { default as Form } from './Form';
export { default as Modal } from './Modal';
export { default as Toast } from './Toast';
export { default as ToastManager } from './Toast/Manager';

export * from './FormFields';
export * from './Stepper';
