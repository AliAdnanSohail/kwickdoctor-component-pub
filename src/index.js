export { default as Button } from './components/Button';
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as Calendar } from './components/Calendar';
export { default as Card } from './components/Card';
export { default as Datepicker } from './components/Datepicker';
export { default as TextInput } from './components/TextInput';
export { default as Toast } from './components/Toast';
<<<<<<< HEAD
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as AvatarPicker } from './components/AvatarPicker';
=======
>>>>>>> master
