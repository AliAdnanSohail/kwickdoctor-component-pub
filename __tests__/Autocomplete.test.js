import React from 'react';
import renderer from 'react-test-renderer';

import { Autocomplete } from '../src';

const suggestions = ['Pseudo', 'classes', 'are', 'bolted', 'on', 'to', 'selectors', 'specify'];

const defaultProps = {
  id: 'autocomplete',
  name: 'Name',
  label: 'Label',
  placeholder: 'Start typing',
  suggestions,
};

it('renders correctly', () => {
  const element = renderer.create(<Autocomplete {...defaultProps} />).toJSON();

  expect(element).toMatchSnapshot();
});
