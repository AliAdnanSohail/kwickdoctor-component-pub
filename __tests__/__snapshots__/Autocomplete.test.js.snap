// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="jsx-1685083184 jsx-3325440485 jsx-107481942 form-field"
>
  <label
    className="jsx-1685083184 jsx-3325440485 jsx-107481942"
    htmlFor="autocomplete"
  >
    Label
  </label>
  <div
    className="jsx-1685083184 jsx-3325440485 jsx-107481942 autocomplete"
    onFocus={[Function]}
    role="textbox"
    tabIndex={0}
  >
    <div
      className="jsx-1685083184 jsx-3325440485 jsx-107481942 autocomplete__input"
    >
      <input
        className="jsx-1685083184 jsx-3325440485 jsx-107481942"
        id="autocomplete"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Start typing"
        type="text"
        value=""
      />
    </div>
  </div>
  <style>
    .autocomplete{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-flex-wrap:wrap;-ms-flex-wrap:wrap;flex-wrap:wrap;position:relative;padding:5px;width:100%;min-width:100%;max-width:100%;min-height:48px;background-color:#ffffff;box-sizing:border-box;border-width:2px;border-style:solid;border-color:#E7E7F0;border-radius:4px;}
    .autocomplete.focused,.autocomplete:focus{outline:none;border-color:#0c97f9;}
    .autocomplete.invalid,.autocomplete.invalid:focus{border-color:#FF5572;}
    .autocomplete__input{position:relative;margin:5px 11px;}
    .autocomplete__input input{line-height:24px;background:transparent;border:none;border-radius:inherit;font-style:normal;font-weight:normal;font-size:0.875rem;}
    .autocomplete__input input::-webkit-input-placeholder{color:#828393;}
    .autocomplete__input input::-moz-placeholder{color:#828393;}
    .autocomplete__input input:-ms-input-placeholder{color:#828393;}
    .autocomplete__input input:-moz-placeholder{color:#828393;}
    .autocomplete__input input:focus{outline:none;}
  </style>
</div>
`;
